### ISSUE TITLE
Empty balance, 1 mined block 

### ISSUE THREAD

**Issue by hjhart** [zcash/zcash#1720]
I can't tell if this is poor reporting on the `zcashd` process or a bug in the mining code, but I'll post my account here for hopefully resolving something.

I was running `zcashd` locally, not as a daemon. After a few minutes, I noticed that I had mined 1 block. Great! Here was the output: 

```
Thank you for running a Zcash node!
You're helping to strengthen the network and contributing to a social good :)

You are running 8 mining threads.

Since starting this node 1 hours, 33 minutes, 6 seconds ago:
- You have validated 7307 transactions!
- You have contributed 15.2444 Sol/s on average to the network solution rate.
- You have completed 45031 Equihash solver runs.
- You have mined 1 blocks!
```

However, no matter where or how I poked at the `zcash-cli` I had no record or indication of that. 

`zcash-cli listunspent` returns an empty array.
`zcash-cli z_getbalance` on my t-wallets return empty as well. 

Now that I've restarted the `zcashd` process I have no record or indication of any mined blocks.

What happened?

**Comment by daira** [comment]
What does ``zcash-cli listtransactions`` show?

**Comment by juscamarena** [comment]
Yes, this was confusing to me as well. It wasn't clear at all. They'll show up in listtransactions and show the balance as immature.. After the 100 block period it will show up in your balance when you do zcash-cli getinfo.... Oh and don't even get me started on having to use zcash-cli sendmany instead of being able to use sendtoaddress like in bitcoin... You also have to send to a z_addr, before being able to send to t_addr. I should probably help with the docs to make it clear for noobs.

**Comment by hjhart** [comment]
```
[19:32:08] james@homebody [~/workspace/zcash]
> ./src/zcash-cli listtransactions
[
]

[19:32:16] james@homebody [~/workspace/zcash]
> ./src/zcash-cli getinfo
{
    "version" : 1000050,
    "protocolversion" : 170002,
    "walletversion" : 60000,
    "balance" : 0.00000000,
    "blocks" : 1301,
    "timeoffset" : 0,
    "connections" : 8,
    "proxy" : "",
    "difficulty" : 43472.29099890,
    "testnet" : false,
    "keypoololdest" : 1477691099,
    "keypoolsize" : 101,
    "paytxfee" : 0.00000000,
    "relayfee" : 0.00005000,
    "errors" : ""
}

```

`listtransactions` was empty yesterday (when the block was mined) and is still empty today. `getinfo` has also always returned a zero balance.

**Comment by juscamarena** [comment]
Go into your debug log and search for "proof" Get the blockhash or the address it mined to and then see if it's in your wallet. Try signing a message with it or something.

**Comment by jamzed** [comment]
I had exactly same situation, for me listtransactions returns empty array as well... debug.log shows this

```
2016-10-28 16:26:21 proof-of-work found
  hash: 000214075b0e00a6a3aa5bda4c7d4eb2307460398be86d4ebe77df378c0b0241
target: 000282c500000000000000000000000000000000000000000000000000000000
2016-10-28 16:26:21 CBlock(hash=000214075b0e00a6a3aa5bda4c7d4eb2307460398be86d4ebe77df378c0b0241, ver=4, hashPrevBlock=0001f254380b99197c1494cdb6e043584c3224ecc21a5a9b25cdf3ce2858782f, hashMerkleRoot=8a6ea3ebff5bba653b222a3958cb58777efa7c92f13d701303a8093f4ad550be, hashReserved=0000000000000000000000000000000000000000000000000000000000000000, nTime=1477671980, nBits=1f0282c5, nNonce=0000b3a436ef49c770ec3360d0b901636ded1f70817d9319a49b9e44c3be0002, vtx=1)
  CTransaction(hash=8a6ea3ebff, ver=1, vin.size=1, vout.size=2, nLockTime=0)
    CTxIn(COutPoint(0000000000, 4294967295), coinbase 014a0110)
    CTxOut(nValue=0.03700000, scriptPubKey=03b07c49cd15071733fb616165f51b)
    CTxOut(nValue=0.00925000, scriptPubKey=OP_HASH160 7d46a730d31f97b1930)

  vMerkleTree:  8a6ea3ebff5bba653b222a3958cb58777efa7c92f13d701303a8093f4ad550be

2016-10-28 16:26:21 generated 0.037
2016-10-28 16:26:21 AddToWallet 8a6ea3ebff5bba653b222a3958cb58777efa7c92f13d701303a8093f4ad550be  new
2016-10-28 16:26:22 AddToWallet 8a6ea3ebff5bba653b222a3958cb58777efa7c92f13d701303a8093f4ad550be
```

**Comment by juscamarena** [comment]
Orphaned. I think.

**Comment by hjhart** [comment]
Okay, it looks like I mined 1 block about 30 seconds after I had started the process. So my logs look like this: 

```
2016-10-28 21:44:58 Loading verifying key from /home/james/.zcash-params/sprout-verifying.key
2016-10-28 21:44:58 Loaded verifying key in 0.000025s seconds.
2016-10-28 21:44:58 Binding RPC on address ::1 port 8232 (IPv4+IPv6 bind any: 0)
2016-10-28 21:44:58 Binding RPC on address 127.0.0.1 port 8232 (IPv4+IPv6 bind any: 0)
...skipping...
2016-10-28 21:45:03 CreateNewBlock(): total size 1000
2016-10-28 21:45:03 Running ZcashMiner with 1 transactions in block (272 bytes)
2016-10-28 21:45:26 ZcashMiner:
2016-10-28 21:45:26 proof-of-work found
  hash: 0000383b1a49354bb1378913b7f5af205e5602bde6aa9b3aa42a11883f8f1f6a
target: 0007ffff00000000000000000000000000000000000000000000000000000000
2016-10-28 21:45:26 CBlock(hash=0000383b1a49354bb1378913b7f5af205e5602bde6aa9b3aa42a11883f8f1f6a, ver=4, hashPrevBlock=0007ddeb44886a3ca9896b912d3ace5ddc8ca0371465dea870994754fbe71eaf, hashMerkleRoot=782007baa45fd78751a5ebef9346655cb86aa5a86f95a77df2d7fe291f94dfe1, hashReserved=0000000000000000000000000000000000000000000000000000000000000000, nTime=1477691125, nBits=1f07ffff, nNonce=0000cf97a2b084eefaf1e5ee26af209de8909c344284a681ae0753e97ab10015, vtx=1)
  CTransaction(hash=782007baa4, ver=1, vin.size=1, vout.size=2, nLockTime=0)
    CTxIn(COutPoint(0000000000, 4294967295), coinbase 600102)
    CTxOut(nValue=0.00800000, scriptPubKey=02e19e1dfac19cfc627ce18de7bd55)
    CTxOut(nValue=0.00200000, scriptPubKey=OP_HASH160 7d46a730d31f97b1930)

  vMerkleTree:  782007baa45fd78751a5ebef9346655cb86aa5a86f95a77df2d7fe291f94dfe1

2016-10-28 21:45:26 generated 0.008
2016-10-28 21:45:26 keypool keep 8
2016-10-28 21:45:26 UpdateTip: new best=0000383b1a49354bb1378913b7f5af205e5602bde6aa9b3aa42a11883f8f1f6a  height=16  log2_work=17.087463  tx=17  date=2016-10-28 21:45:25 progress=1.000000  cache=0.0MiB(16tx)
2016-10-28 21:45:26 AddToWallet 782007baa45fd78751a5ebef9346655cb86aa5a86f95a77df2d7fe291f94dfe1  new
```

My current understanding is that I obtained an orphaned block. I found an article explaining those better here: https://bitcoin.stackexchange.com/questions/5859/what-are-orphaned-and-stale-blocks

For future reference, are there bits of my debug.log that I shouldn't be pasting in a public forum?

**Comment by juscamarena** [comment]
Nothing really. Unless you were under a different nick and didn't want anyone knowing you were James. /home/james.

**Comment by hjhart** [comment]
Thanks for your help @juscamarena. I just looked through the zchain explorer and found this transaction, which occurred nearly fifteen minutes after my proof-of-work: https://explorer.zcha.in/transactions/176d985759ee395109c8eee0defa43d36d655d1e2d1ee685da0b8f9319f4ec18

(Mine happened at 21:45:26 UTC, which, (correct my time zone math if I'm wrong), would be 14:45:26 PT)

Forgive my ignorance, as I'm new to this, but how does this possibly happen?

![screen shot 2016-10-30 at 12 28 51 am](https://cloud.githubusercontent.com/assets/547981/19835069/34e38940-9e38-11e6-9428-683a18959152.png)

**Comment by juscamarena** [comment]
I believe you found a block before syncing up to the rest of the network see how the height is still height=16, but at that time the block height was actually much higher in the 570s.

**Comment by ryandotsmith** [comment]
I also mined a block that was subsequently re-organized by the network. I was super excited when I saw the message in my logs only to be disappointed when I checked my wallet's balance.

Perhaps the "You have mined 1 blocks!" message can be a bit more conservative?

**Comment by hjhart** [comment]
Right – there is quite a bit of initial excitement related to this message, followed by a bit of confusion (where are my coins?), and then disappointment now when I realize I haven't actually mined any blocks.

I think in this thread alone we've already got three confirmed cases, I imagine more incoming.

I'm not sure what the roadmap looks like for zcash at this point, (I imagine there are high priority bugs / features on the third day after release), but it'd be good to hear from someone within the zcash orginization as to what direction they think we should go.

**Comment by hjhart** [comment]
I believe this issue might be relevant: https://github.com/zcash/zcash/issues/1716

**Comment by juscamarena** [comment]
Yeah, basically the miner starts mining with the info it currently has, so if it doesn't know the longest chain yet it can potentially mine a block on what it thinks is the tip of the chain. You can't really get around that, but you can like in that issue add detection for blocks mined that don't make it on the main chain for the info screen to not confuse new users that they may have found a block when they haven't.

**Comment by str4d** [comment]
I have implemented a fix for #1716 in #1735, in which I both detect orphans and count the actual revenue. It should be sufficient to resolve this issue.

Closing this because the OP's problem has been diagnosed, and the next step is already tracked in #1716.
